libs/providers/config-cat/src/lib/config-cat-provider.ts

```diff
@@ -33,50 +39,37 @@ export class ConfigCatProvider implements Provider {
     return new ConfigCatProvider(...params);
   }
 
-  get status() {
-    return this._status;
-  }
-
   public async initialize(): Promise<void> {
-    return new Promise((resolve) => {
-      const originalParameters = this._clientParameters;
-      originalParameters[2] ??= {};
-
-      const options = originalParameters[2];
-      const oldSetupHooks = options.setupHooks;
-
-      options.setupHooks = (hooks) => {
-        oldSetupHooks?.(hooks);
-
-        // After resolving, once, we can simply emit events the next time
-        const onProviderReady = () => {
-          this.events.emit(ProviderEvents.Ready);
-          this._status = ProviderStatus.READY;
-        };
-
-        hooks.once('clientReady', () => {
-          hooks.on('clientReady', onProviderReady);
-          onProviderReady();
-          resolve();
-        });
+    const originalParameters = this._clientParameters;
+    originalParameters[2] ??= {};
 
-        hooks.on('configChanged', (projectConfig: IConfig | undefined) =>
-          this.events.emit(ProviderEvents.ConfigurationChanged, {
-            flagsChanged: projectConfig ? Object.keys(projectConfig.settings) : undefined,
-          }),
-        );
-
-        hooks.on('clientError', (message: string, error) => {
-          this._status = ProviderStatus.ERROR;
-          this.events.emit(ProviderEvents.Error, {
-            message: message,
-            metadata: error,
-          });
+    const options = originalParameters[2];
+    const oldSetupHooks = options.setupHooks;
+
+    options.setupHooks = (hooks) => {
+      oldSetupHooks?.(hooks);
+
+      hooks.on('clientReady', () => {
+        this.events.emit(ProviderEvents.Ready);
```

Comment:
This may result in duplicate READY events, since the SDK will emit one when this function completes, and also due to this handler.

@kinyoklion opened an issue discussing this [here](https://github.com/open-feature/spec/issues/265).

Reply:
Makes sense, how do we continue on this @toddbaert?

Reply:
We can either implement something as @kinyoklion describes or we can just add some logic to prevent the emissions of the initial event.